meps/mep-0003.md

```diff
@@ -0,0 +1,257 @@
+
+## UX implications & migration
+
+The proposed syntax is an addition to existing role and directive patterns. It is a stricly non-breaking enhancement; existing MyST documents that use this syntax are currently considered invalid and should not successfully parse. It does not invalidate or change the behavior of any existing workflows.
+
+The inline attribute syntax adopts the same inline attribute syntax used by Pandoc, Quarto, and djot, with the exception that the inline attribute must start with the role/directive name. This change follows from adapting the reference syntax to MyST Markdown, which already implements extensibilty through nominal types (directives and roles) rather than compositional types (i.e. class names).
+
+## Alternatives Considered
+
+We considered other approaches in this proposal and in discussion over several years. Including:
+
+1. extending the patterns used in Sphinx internal to the role content (`` {name}`content <argument .class1 key="value">` ``)
+2. attributes specified _after_ a role in addition to the name (``{name}`content`{.class1 key="value"}``); and
+3. bracketed spans `[content]{.class}`
+
+We chose the current approach to (a) separate the options and content clearly (i.e. not 1); (b) to keep role names and options close together (i.e. not 2); and \(c) to leave future syntax options open (i.e. 3 is a possible future addition).
+
+### Future Syntax Possibilities
+
+(inline-attrs-elements)=
+
+#### Inline Attributes for Elements
+
+The `button` role uses a Sphinx-style link DSL to embed both a link description and link target into the button. If instead if were possible to use (unnamed) inline-attributes specification, instead of writing
+
+```
+{button}`See this link <https://foo.com>`
+```
+
+one could write
+
+```
+[See this link](https://foo.com){.button}
+```
+
+This syntax would also make it easier to annotate e.g. CommonMark images with widths:
+
+```
+![](https://foo.com/image){width=512px}
+```
+
+#### Anonymous Roles and Directives
+
+Perhaps in conflict with [](#inline-attrs-elements), we could provide a short-hand for the div and span roles, where
+
+````
+```{div .red}
+Some markup
+```
+````
+
+becomes
+
+````
+```{.red}
+Some markup
+```
+````
+
+(bool-attrs)=
+#### Short-hand Boolean Attibutes
+
+In certain markup languages, such as React JSX, it is possible to pass boolean `true` values simply by name, e.g.
+
+```
+<FooComponent myProp />
+```
+
+where `props.myProp` will be `true`. We might consider extending this syntax such that:
+
+````
+```{code linenos}
```

Comment:
flag values are not part of pandoc/djot: https://github.com/jgm/djot/issues/257

Reply:
I added a section note to explicitly say this is out of scope of this MEP.

> These syntax possibilities are _not_ suggested in this MEP, however, they came up in discussions for possible future syntax directions.